src/index.js

```diff
@@ -0,0 +1,123 @@
+      </li>
+    `;
+  });
+
+  const addedTasks = document.querySelectorAll('.task');
+
+  const checkboxContainers = document.querySelectorAll('.task > .checkbox-container > input[type="checkbox"]');
+
+  checkboxContainers.forEach((checkbox) => {
+    const inputText = checkbox.nextElementSibling;
+    let previousState = checkbox.checked;
+
+    inputText.readOnly = true;
+
+    checkbox.addEventListener('change', (event) => {
+      const currentState = event.target.checked;
+
+      if (currentState !== previousState) {
+        const foundTask = tasks.find((task) => task.description === inputText.value);
+        if (foundTask) {
+          foundTask.completed = currentState;
+          updateStatus(tasks.indexOf(foundTask), currentState);
+        }
+      }
+
+      previousState = currentState;
+    });
+  });
+
+  addedTasks.forEach((task, index) => {
+    const textInput = task.querySelector('input[type="text"]');
+    task.addEventListener('click', () => {
+      textInput.readOnly = false;
+      if (task.querySelector('.fa-ellipsis-vertical')) {
+        const ellipsisIcon = task.querySelector('.fa-ellipsis-vertical');
+        ellipsisIcon.classList.remove('fa-ellipsis-vertical');
+        ellipsisIcon.classList.add('fa-trash');
+        ellipsisIcon.addEventListener('click', () => {
+          deleteTask(index);
+          displayTasks();
+        });
+      } else {
+        const trashIcon = task.querySelector('.fa-trash');
+        trashIcon.classList.remove('fa-trash');
+        trashIcon.classList.add('fa-ellipsis-vertical');
+        textInput.readOnly = true;
+      }
+    });
+
+    // Edit
+    textInput.addEventListener('input', () => {
+      const data = textInput.value.trim();
+      editTask(data, index);
+    });
+  });
+};
+
+const initializeTasks = () => {
+  document.addEventListener('DOMContentLoaded', displayTasks);
+};
+
+const refreshPage = () => {
+  localStorage.removeItem('Tasks');
+  window.location.reload();
+};
+
+clearBtn.addEventListener('click', () => {
+  clearAllCompletedTasks();
+  sortTasks();
+  storeTasksToLocalStorage();
+  displayTasks();
+});
+
+initializeTasks();
+addBtn.addEventListener('click', () => {
+  const inputField = document.getElementById('input-task');
+  const taskDescription = inputField.value.trim();
+
+  // Check if the task description is not empty
+  if (taskDescription !== '') {
+    addTask(taskDescription);
+    inputField.value = '';
+    displayTasks();
+  }
+});
+document.addEventListener('keypress', (e) => {
+  if (e.key === 'Enter') {
+    const inputField = document.getElementById('input-task');
+    const taskDescription = inputField.value.trim();
+
+    // Check if the task description is not empty
+    if (taskDescription !== '') {
+      addTask(taskDescription);
+      inputField.value = '';
+      displayTasks();
+    }
+  }
+});
+document.querySelector('.fa-arrows-rotate').addEventListener('click', refreshPage);
+
+export {};
```

Comment:
Kindly remove the line "export {};" at the end of the file, as it is not necessary.